File: BenchMarks/BM.lua
chat:log("lua start time: " .. time:time())
local prStart = time:time()
chat:log("  Prime (2000) - Start: " .. prStart)
do
	local primes = {}
	local num = 2
	while #primes < 2000 do
		local flag = true
		for _,prime in pairs(primes) do
			if (num % prime) == 0 then
				flag = false
				break
			end
		end
		if flag then
			table.insert(primes, num)
		end
		num = num + 1
	end
end
local prEnd = time:time()
chat:log("  Prime (2000) - End: " .. prEnd .. ", duration: ".. (prEnd-prStart))
local gbStart = time:time()
chat:log("  getBlock (4 full chunk) - Start: " .. gbStart)
do
	local blocks = {}
	local pla = player:getPlayer()
	local chunk = {pla:getX() - (pla:getX() % 16), pla:getZ() - (pla:getZ() % 16)}
	for x = chunk[1], chunk[1]+31 do
		for z = chunk[2], chunk[2]+31 do
			for y = 0, 255 do
				table.insert(blocks, world:getBlock(x,y,z):getId())
			end
		end
	end
end
local gbEnd = time:time()
chat:log("  Prime (2000) - End: " .. gbEnd .. ", duration: ".. (gbEnd-gbStart))

File: BenchMarks/BM.js
chat.log(`js start time: ${Date.now()}`)
const prStart = Date.now()
chat.log(`  Prime (2000) - Start: ${prStart}`)
{
	const primes = []
	let num = 2
	while (primes.length < 2000) {
		let flag = true
		for (const prime of primes) {
			if (!(num % prime)) {
				flag = false
				break
			}
		}
		if (flag) {
			primes.push(num)
		}
		++num
	}
}
const prEnd = Date.now()
chat.log(`  Prime (2000) - End: ${prEnd}, duration: ${prEnd-prStart}`)
const gbStart = Date.now()
chat.log(`  getBlock (4 full chunk) - Start: ${gbStart}`)
{
	const blocks = []
	const pla = player.getPlayer()
	const chunk = [pla.getX() >> 4, pla.getZ() >> 4]
	const start = [chunk[0] << 4, chunk[1] << 4]
	const end = [start[0] + 32, start[1] + 32]
	for (let x = start[0]; x < end[0]; ++x) {
		for (let z = start[1]; z < end[1]; ++z) {
			for (let y = 0; y < 256; ++y) {
				blocks.push(world.getBlock(x,y,z).getId())
			}
		}
	}
}
const gbEnd = Date.now()
chat.log(`  getBlock (4 full chunk) - End: ${gbEnd}, duration: ${gbEnd-gbStart}`)